Mathlib/Order/SupClosed.lean

```diff
@@ -79,6 +79,16 @@ lemma supClosed_pi {ι : Type*} {α : ι → Type*} [∀ i, SemilatticeSup (α i
     {t : ∀ i, Set (α i)} (ht : ∀ i ∈ s, SupClosed (t i)) : SupClosed (s.pi t) :=
   fun _a ha _b hb _i hi ↦ ht _ hi (ha _ hi) (hb _ hi)
 
+lemma SupClosed.insert_upper_bound {s : Set α} {a : α} (hs : SupClosed s) (ha : ∀ b ∈ s, b ≤ a) :
```

Comment:
Should `ha` be phrased in terms of`upperBounds s`? Idem dito for the statements below.

Reply:
Yes -done thanks. Curiously `aesop` needed a hint one way around but not the other.